### PR TITLE
Support for path to the private key file for the "private_key" argument

### PR DESCRIPTION
**Pull request by Kami**
Note: This PR is based on #2666.

This pull request updates remote runner so user can now also specify a path to the private key file for the `private_key` remote runner argument.

This resolves security concerns from #2662.

We are currently working on secrets in datastore story which goal is to solve some of the issues like that, but I still see value in allowing users to specify a path to the key file. A lot of organizations already have a code / configuration management in place or managing secret files on disk so it makes sense to allow users to re-use that functionality instead of forcing them to use datastore secrets which is StackStorm specific functionality.

In addition to that, I also fixed support for password protected keys (see https://github.com/StackStorm/st2/pull/2596#issuecomment-216768910 for details) and removed support for the deprecated `key` argument since the whole logic for that was kinda convoluted and made it hard to follow / understand.

## TODO:

- [x] Docs
- [x] Tests

### CODE REVIEW

**Comment by Kami** [comment]
I've added some tests for `ParallelSSHClient` instantiation inside the runner `pre_run` method - a9df32e701b7e9518055141cc5886f50bbe8e072

Previously we had no tests for this code path so we didn't tests the runner argument passing end to end (we only tested ParallelSSHClient and ParamikoSSHClient independently) which means runner could be broken end to end and our tests wouldn't catch that.

That's also the reason why we didn't catch that key passpharse support never worked in the first place - we didn't have any tests for that code path.

File: st2common/st2common/bootstrap/runnersregistrar.py [hunk 1]
```diff
@@ -134,8 +134,8 @@
                 'secret': True
             },
             'private_key': {
-                'description': ('Private key material to log in. Note: This needs to be actual '
-                                'private key data and NOT path.'),
+                'description': ('Private key material or path to the private key file on disk '
```
Comment (lakshmi-kannan) [inline st2common/st2common/bootstrap/runnersregistrar.py]: Depending on what user the action runner runs as, this is a huge problem without documentation. If the action runner process runs as ``st2``, st2 needs to be able to read this key. There should be docs around the risk of making the key readable by st2. Without a corrresponding change to deprecate core.local, this IMO is a huge issue. I am -1 to supporting path to key file.

Comment (Kami) [inline st2common/st2common/bootstrap/runnersregistrar.py]: I agree about the risk and the need for the documentation on it.

Comment (Kami) [inline st2common/st2common/bootstrap/runnersregistrar.py]: In any case, I will add some docs on the limitations and security implications of this approach.

Then we can decide if we are fine with it or not. If we are not, I will open a new PR and at least cherry-pick some fixes, improvements and tests from this PR.

Comment (Kami) [inline st2common/st2common/bootstrap/runnersregistrar.py]: Added docs which clarify the requirements and potential security risks when using this feature - https://github.com/StackStorm/st2docs/pull/147.

**Comment by Kami** [comment]
I added the docs in https://github.com/StackStorm/st2docs/pull/147 so I will go ahead and merge this.
